### PR TITLE
Feature/hir 129 implement invitation sending

### PR DESCRIPTION
**Pull request by AlexeyShok**
Added endpoints for invitation managers. Resend implemented too.

### CODE REVIEW

File: HappyTravel.Hiroshima.DirectManager/Models/Requests/SendManagerInvitation.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace HappyTravel.Hiroshima.DirectManager.Models.Requests
+{
+    public class SendManagerInvitation : ManagerInfo
```
Comment (hasmikharutyunyan) [inline HappyTravel.Hiroshima.DirectManager/Models/Requests/SendManagerInvitation.cs:5]: I think models must be a noun, not a verb

File: HappyTravel.Hiroshima.WebApi/Controllers/DirectManager/ManagerController.cs [hunk 2]
```diff
@@ -134,6 +136,78 @@ async Task<Result<string>> GetEmailFromIdentity()
         }
 
 
+        /// <summary>
+        ///     Invite regular manager by e-mail.
```
Comment (oleg1std) [inline HappyTravel.Hiroshima.WebApi/Controllers/DirectManager/ManagerController.cs:140]: Sends an invite to a regular manager

File: HappyTravel.Hiroshima.WebApi/Controllers/DirectManager/ManagerController.cs [hunk 3]
```diff
@@ -134,6 +136,78 @@ async Task<Result<string>> GetEmailFromIdentity()
...
+        /// </summary>
+        /// <param name="managerInvitation">Regular manager registration request.</param>
+        /// <returns></returns>
+        [HttpPost("invitations/send")]
+        [ProducesResponseType((int)HttpStatusCode.NoContent)]
+        [ProducesResponseType(typeof(ProblemDetails), (int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> InviteMaanager([FromBody] Hiroshima.DirectManager.Models.Requests.SendManagerInvitation managerInvitation)
```
Comment (oleg1std) [inline HappyTravel.Hiroshima.WebApi/Controllers/DirectManager/ManagerController.cs:147]: Typo

File: HappyTravel.Hiroshima.WebApi/Controllers/DirectManager/ManagerController.cs [hunk 4]
```diff
@@ -134,6 +136,78 @@ async Task<Result<string>> GetEmailFromIdentity()
...
+
+
+        /// <summary>
+        ///    Resend manager invitation
+        /// </summary>
+        /// <param name="invitationCode">Invitation code</param>>
+        [HttpPost("invitations/{invitationCode}/resend")]
+        [ProducesResponseType((int)HttpStatusCode.NoContent)]
+        [ProducesResponseType(typeof(ProblemDetails), (int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> Resend([FromRoute] string invitationCode)
```
Comment (oleg1std) [inline HappyTravel.Hiroshima.WebApi/Controllers/DirectManager/ManagerController.cs:164]: ResendInvite

File: HappyTravel.Hiroshima.DirectManager/Services/ManagerInvitationService.cs [hunk 5]
```diff
@@ -1,20 +1,240 @@
...
+            }
+
+
+            async Task<Result<ManagerInvitation>> SendInvitationMail(ManagerInvitation invitationData)
+            {
+                var serviceSupplier = await _managerContext.GetServiceSupplier();
+                if (serviceSupplier.IsFailure)
+                    return Result.Failure<ManagerInvitation>(serviceSupplier.Error);
+
+                //await _notificationService.SendInvitation(managerInvitationInfo, serviceSupplier.Value.Name);
```
Comment (mingazhev) [inline HappyTravel.Hiroshima.DirectManager/Services/ManagerInvitationService.cs:141]: Need to add a comment, why is commented, when will be uncommented.

File: HappyTravel.Hiroshima.DirectManager/Services/ManagerInvitationService.cs [hunk 6]
```diff
@@ -1,20 +1,240 @@
...
+
+                return invitationData;
+            }
+
+
+            async Task<Result> DisableExistingInvitation(ManagerInvitation existingInvitation)
+            {
+                if (existingInvitation is null)
```
Comment (mingazhev) [inline HappyTravel.Hiroshima.DirectManager/Services/ManagerInvitationService.cs:149]: Is this  possible?

File: HappyTravel.Hiroshima.DirectManager/Services/ManagerInvitationService.cs [hunk 7]
```diff
@@ -1,20 +1,240 @@
...
+                _dbContext.Update(managerInvitation);
+                await _dbContext.SaveChangesAsync();
+            }
+        }
+
+
+        public Task<Result<Models.Responses.ManagerInvitation>> GetPendingInvitation(string invitationCode)
+        {
+            return GetInvitation(invitationCode).ToResult("Could not find invitation")
+                .Ensure(IsNotAccepted, "Already accepted")
```
Comment (mingazhev) [inline HappyTravel.Hiroshima.DirectManager/Services/ManagerInvitationService.cs:178]: Need to add more words to make message more descriptive

File: HappyTravel.Hiroshima.DirectManager/Services/ManagerInvitationService.cs [hunk 8]
```diff
@@ -1,20 +1,240 @@
...
+                    managerInvitation.Title,
+                    managerInvitation.Position,
+                    managerInvitation.Email,
+                    managerInvitation.ManagerId,
+                    managerInvitation.ServiceSupplierId);
+            }
+        }
+
+
+        private Result<(ManagerServiceSupplierRelation, string)> GenerateInvitationCode(ManagerServiceSupplierRelation managerInvitation)
```
Comment (mingazhev) [inline HappyTravel.Hiroshima.DirectManager/Services/ManagerInvitationService.cs:207]: Do we need a result there? No failure can occur

**Review by hasmikharutyunyan — APPROVED** [review]

File: HappyTravel.Hiroshima.DirectManager/Services/ManagerInvitationService.cs (hunk 8 above)
Comment (AlexeyShok) [reply HappyTravel.Hiroshima.DirectManager/Services/ManagerInvitationService.cs:204]: It's need for pipeine

**Pull request merged**